docs/UserGuide.md

```diff
@@ -44,6 +44,70 @@ SocialBook is a **desktop app for managing contacts, optimized for use via a  Co
 
 --------------------------------------------------------------------------------------------------------------------
 
+## Contact field requirements
+
+### Name
+* Names are compulsory for all contacts, and are denoted with the `n/` prefix.
+* Names can contain any characters at all, including spaces, hyphens, and other special characters.
+* Names will be stored in their case-sensitive form, but capitalisation will be ignored when checking for duplicate names.
+    * Eg. Adding a contact as "john Doe" will save them as such, but trying to add a "John Doe" with the same phone number will be marked as a duplicate person and rejected.
+    * To avoid unexpected behaviour with this, it is recommended that users save contacts with consistent capitalisation rules.
+
+### Phone
+* Phones are compulsory for all contacts, and are denoted with the `p/` prefix.
+* Phone numbers can only contain 8 numbers, and must begin with a 6, 8, or 9.
+* Spaces in the middle of a phone number are accepted (eg. 9123 4523), as are phone numbers without spaces (eg. 91234523).
+* Spaces in unusual locations will render the phone number invalid (eg. 912 34523).
+
+### Address
+* Addresses are optional for contacts, and are denoted by the `a/` prefix.
+* Addresses can contain any characters, including spaces, commas, hyphens, etc.
+* To indicate no address for a contact, you can `add` a contact without the `a/` prefix, or with a `a/` followed by whitespace.
+
+### Email
+* Emails are optional for contacts, and are denoted by the `e/` prefix.
+* Email addresses are confined to the limits of the traditional email format: **`localPart@domain.label`**. This includes a few restrictions:
+    * The `localPart` and `domain` components of the email must be alphanumeric, with no special characters.
+    * The `localPart` and `domain` components of the email must be separated by a `@`.
+    * The `label` component must be alphanumeric, and contain at least 2 characters.
+    * The `domain` and `label` component must be separated by a `.`.
+* To indicate no email for a contact, you can `add` a contact without the `e/` prefix, or with a `e/` followed by whitespace.
+
+### Date of Last Visit
+* Dates of last visit are optional for contacts, and are denoted by the `d/` prefix.
+* Dates of last visit are confined to the `DD-MM-YYYY` format.
```

Comment:
should we change this to `dd-MM-yyyy` like in the gui message for consistency? since we can't change what is on the app.

Reply:
Yeah can I think rayner's PR does this, but let me double check.